### PR TITLE
Auto split long images to improve performance of readers

### PR DESCRIPTION
**Pull request by saud-97**
Added a feature to optimize downloaded chapters reading performance and stability 
Fixes: https://github.com/tachiyomiorg/tachiyomi/issues/6973 https://github.com/Jays2Kings/tachiyomiJ2K/issues/1211

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/data/download/Downloader.kt [hunk 1]
```diff
@@ -496,6 +497,69 @@ class Downloader(
 
         // Only rename the directory if it's downloaded.
         if (download.status == Download.State.DOWNLOADED) {
+            // Implementation of Auto Split long images upon download.
+            if (preferences.splitLongImages().get()) {
+                for (curPage in downloadedImages) {
+                    // Checking the image dimensions without loading it in the memory.
+                    val options = BitmapFactory.Options()
```
Comment (FourTOne5) [inline app/src/main/java/eu/kanade/tachiyomi/data/download/Downloader.kt:504]: Should just move the whole stuff to ImageUtil

Comment (saud-97) [reply app/src/main/java/eu/kanade/tachiyomi/data/download/Downloader.kt:507]: I'm not really that good in kotlin and don't know how to do it in `ImageUtil`. If you can do the changes you would do me a favor. But is there any real benefits from switching?

**Comment by saud-97** [comment]
I noticed that the loader doesn't change the images source until someone leaves the manga page and then return back in. 

To explain this: if someone read a few pages in a chapter then went back to manga page and downloaded the chapter. After he enters the chapter again he will still see the cached images not the downloaded (and splitted) one. Would appreciate if someone could fix that loader thing.

**Review by Riztard — APPROVED** [review]

File: app/src/main/java/eu/kanade/tachiyomi/data/download/Downloader.kt [hunk 2]
```diff
@@ -34,8 +36,7 @@ import rx.android.schedulers.AndroidSchedulers
 import rx.schedulers.Schedulers
 import rx.subscriptions.CompositeSubscription
 import uy.kohesive.injekt.injectLazy
-import java.io.BufferedOutputStream
-import java.io.File
+import java.io.*
```
[the remark this answers is not in the document]
Comment (saud-97) [reply app/src/main/java/eu/kanade/tachiyomi/data/download/Downloader.kt]: I have removed it in local, not sure if the commit got removed or not yet.

File: app/src/main/java/eu/kanade/tachiyomi/ui/setting/SettingsDownloadController.kt [hunk 3]
```diff
@@ -16,16 +16,7 @@ import eu.kanade.tachiyomi.data.database.DatabaseHelper
...
-import eu.kanade.tachiyomi.util.preference.defaultValue
-import eu.kanade.tachiyomi.util.preference.entriesRes
-import eu.kanade.tachiyomi.util.preference.intListPreference
-import eu.kanade.tachiyomi.util.preference.multiSelectListPreference
-import eu.kanade.tachiyomi.util.preference.onClick
-import eu.kanade.tachiyomi.util.preference.preference
-import eu.kanade.tachiyomi.util.preference.preferenceCategory
-import eu.kanade.tachiyomi.util.preference.switchPreference
-import eu.kanade.tachiyomi.util.preference.titleRes
+import eu.kanade.tachiyomi.util.preference.*
```
[the remark this answers is not in the document]
Comment (saud-97) [reply app/src/main/java/eu/kanade/tachiyomi/ui/setting/SettingsDownloadController.kt:19]: I have removed it in local, not sure if the commit got removed or not yet.

**Pull request closed without merging**